src/contracts/interfaces/IClaimingManager.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @title Interface for the `IClaimingManager` contract.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ */
+interface IClaimingManager {
+
+    /// STRUCTS ///
+
+    struct DistributionRoot {
+        uint32 activatedAfter; // timestamp after which the root can be claimed against
+        bytes32 root; // merkle root of the distribution
+    }
+
+    struct PaymentMerkleClaim {
+        IERC20 token; 
+        uint256 amount;
+        address recipient; // Explicit recipient of the claim
+
+        uint32 rootIndex; // The index of the root in the list of roots for this token
+        uint32 leafIndex; // The index of the leaf in the merkle tree for this root
+
+        bytes proof;
+    }
+
+    /// EVENTS /// 
+
+    event PaymentUpdaterSet(address oldPaymentUpdater, address newPaymentUpdater);
+    event ActivationDelaySet(uint32 oldActivationDelay, uint32 newActivationDelay);
+    event GlobalCommissionBipsSet(uint16 oldGlobalCommissionBips, uint16 newGlobalCommissionBips);
+    event RecipientSet(address account, address recipient);
+    event RootSubmitted(bytes32 root, uint32 paymentsCalculatedUntilTimestamp, uint32 activatedAfter);
+    event PaymentClaimed(IERC20 token, address recipient, uint256 amount);
+    
+    /// VIEW FUNCTIONS ///
+
```

Comment:
a couple convenience functions + struct which I could see us adding:
```
struct MerkleLeaf {
        IERC20 token; 
        uint256 amount;
        address recipient;
}
``` 
^MerkleLeaf could then be used inside the `PaymentMerkleClaim` struct definition
```
function findLeafHash(MerkleLeaf calldata leaf) external view returns (bytes32)
function checkClaim(PaymentMerkleClaim calldata claim) external view returns (bool)
```

these might be overkill, but I could see them being somewhat useful, at least.